### ISSUE TITLE
Changes feed not returning the latest revision for few docs

### ISSUE THREAD

**Issue by ashvindersingh** [couchbase/sync_gateway#1262]
Scenario: 
- From admin api add a single user
- create 100 docs
- create 10 revisions for each doc
- get changes feed
  Some documents do not have the latest revision and doc contents. Out of 100 docs ~ 2 docs have incorrect revision. 1 less revision than expected. 
  Querying (doing a get on) the doc-id provides the correct revision and doc contents

Setup: 
CBS: 3.1.1 (2 Node cluster)
SG: Dist-index branch

**Comment by ashvindersingh** [comment]
Attached logs: https://issues.couchbase.com/browse/CBLT-389

**Comment by adamcfraser** [comment]
Based on the attached logs, there are 1100 entries seen on the DCP feed, and then 1100 entries written to the index for [channel-1].  So it appears that all the revisions are being written to the index.

It would help to see the _changes response, and get the details on which docs aren't showing the expected revision number.

**Comment by ashvindersingh** [comment]
Please see http://172.23.105.68:4984/db/_changes
Doc-ID: test-54 and doc-id: test-50 are not showing the correct revisions. 
showing rev: 10, expecting rev:11

**Comment by adamcfraser** [comment]
Additional observations:
- index entry document exists for the omitted revision in the index bucket:

```
_idx_entry:77:11
{
  "Sequence": 11,
  "DocID": "test-54",
  "RevID": "11-c59eba38b6ff48aea30c74606e8e4885",
  "Flags": 0,
  "VbNo": 77,
  "TimeSaved": "2015-11-03T17:49:58.762238197-08:00",
  "TimeReceived": "2015-11-03T17:48:52.285950752-08:00",
  "Channels": null,
  "Skipped": false,
  "IsPrincipal": false
}

```
- SG logs include the following for the docs missing revisions (and only those two docs)

```
CRUD+: Got old revision "test-54" / "10-e0419f6395e54da6e886b3de37f0a1ea" --> 38 bytes
CRUD+: Got old revision "test-50" / "10-e0419f6395e54da6e886b3de37f0a1ea" --> 38 bytes
```
- Channel clock for both channel-1 and \* has the correct sequence for the vbucket: [77:11]
- Stable clock also has the correct sequence for the vbucket: [77:11]

**Comment by adamcfraser** [comment]
SG logs include the following for the docs missing revisions (and only those two docs)

CRUD+: Got old revision "test-54" / "10-e0419f6395e54da6e886b3de37f0a1ea" --> 38 bytes
CRUD+: Got old revision "test-50" / "10-e0419f6395e54da6e886b3de37f0a1ea" --> 38 bytes

**Comment by adamcfraser** [comment]
The index blocks for vb 77 (_idx:*:block0:2679, _idx:channel-1:block0:2679) don't include 77:11 - the last sequence in the index block for the vbucket is 77:10.

**Comment by adamcfraser** [comment]
@tleyden This feels like it may be related to SetBulk.  I don't see anything in the error logs about failed block updates, but I'm wondering whether the error isn't being captured/returned from writeBlockSetsWithCas.

**Comment by tleyden** [comment]
@adamcfraser ok as a first pass, I will add more verbose error logging and have @ashvindersingh retest.

**Comment by tleyden** [comment]
Note to self: here's a capture of the _changes feed

https://gist.github.com/112773fd7e769925439e

Hopefully it's in the same state as in the [previous comment](https://github.com/couchbase/sync_gateway/issues/1262#issuecomment-153615982).

**Comment by sethrosetter** [comment]
This run was was targeting feature/distributed_index_bulk_set, build number 346. It will probably provide some clues to the above issue. http://qa.sc.couchbase.com/view/mobile/job/centos-syncgateway-functional-tests-distributed-index/71/. Logs here - https://s3.amazonaws.com/sync-gateway-logs/2015-11-05-09-26-42-sglogs.zip

**Comment by zgramana** [comment]
Should be resolved by #1247.
